architecture/2024-08-applications-rp-component-threat-model.md

```diff
@@ -0,0 +1,300 @@
+
+**Mitigation**
+
+Radius should add validation to make sure the incoming request is from UCP. 
+
+**Status**
+
+Pending
+
+#### Sniffing the communication between Applications RP  and Kubernetes API Server / UCP could cause information disclosure 
+
+**Description:** 
+If a malicious actor could sniff communication between the applications RP and the UCP, the actor could modify/replay the packets.
+
+**Impact:** 
+A malicious actor could use the information about the resources and operations in progress. They can also replay the same requests to cause a DDoS or send a modified payload to cause inadvertent changes to application.
+
+**Mitigations**
+
+1. Tampering with Application RP code/ configs would require access to modify the `radius-system` namespace. Our threat model assumes that the operator has limited access to the `radius-system` namespace using Kubernetes' existing RBAC mechanism. It would be challenging for an arbitrary user to gain access to the cluster and capture traffic.
+
+2. We should add support in Radius to use mTLS as communication protocol between UCP and Applications RP.
+
+**Status:** 
+
+1. Active. All mitigations listed are currently active. Operators are expected to secure their cluster and limit access to the `radius-system` namespace.
+
+2. Pending. We should add support in Radius to use mTLS as communication protocol between UCP and Applications RP.
+
+#### Using recipes can cause escalation of privilege
+
+**Description** 
+A malicious user could become the admin, and configure recipes to create / update resources without having permissions to access these resources directly.
+
+**Impact** 
+This can facilitate the attackers to create resources based on any arbitrary image. 
+
+**Mitigations**
+1. Ability to register recipes should be given to only trusted employees. This in will be possible once Radius to support RBAC. 
+
+2. Since each application resource is deployed in the application's namespace, we can use Kubernetes RBAC to confine potential security implications to the application's namespace.
+
+3. Any container launched by Radius is subject to Kubernetes as the platform. Users should configure their Kubernetes cluster to ensure that pods are created only from images hosted on trusted URLs or registries.
+
+4. Users should configure RBAC of Radius with principle of least privilege so that blast radius of any security incidents involving Radius is minimized.
+
+
+**Status**
+1. Pending. We should implement Radius RBAC and then use the feature to restrict recipe registration to only trusted employees.
+   
+2. Active. Operators are expected limit access so that an application in its own namespace cannot by affected by application in another namespace.
+   
+3. Active. Operators are expected to configure their Kubernetes cluster with trusted URLs or registries.
+   
+4. Active. Operators are expected to configure RBAC of Radius appropriately to minimize its blast radius.
```

Comment:
There is no authorization support in Radius today, so this isn't an active mitigation.